com.unity.multiplayer.mlapi/Tests/Editor/NetworkBehaviourTests.cs

```diff
@@ -0,0 +1,59 @@
+using System;
+using NUnit.Framework;
+using UnityEngine;
+using Object = UnityEngine.Object;
+
+namespace MLAPI.EditorTests
+{
+    public class NetworkBehaviourTests
+    {
+        [Test]
+        public void HasNetworkObjectTest()
+        {
+            var gameObject = new GameObject(nameof(NetworkObject));
+            var networkBehaviour = gameObject.AddComponent<EmptyNetworkBehaviour>();
+            
+            Assert.That(networkBehaviour.HasNetworkObject, Is.False);
+
+            var networkObject = gameObject.AddComponent<NetworkObject>();
+            
+            Assert.That(networkBehaviour.HasNetworkObject, Is.True);
+            
+            Object.DestroyImmediate(networkObject);
+            
+            Assert.That(networkBehaviour.HasNetworkObject, Is.False);
```

Comment:
wouldn't `NetworkBehaviour` instance also be destroyed as part of `NetworkObject`/`GameObject` we created above after we `DestroyImmediate()`?

Reply:
No? We just remove the NetworkObject component.

Reply:
oh I see, then we should also destroy the main GameObject either within this test method or on `[TearDown]` :)